packages/studio-web/src/app/demo/demo.component.ts

```diff
@@ -35,6 +35,29 @@ export class DemoComponent implements OnInit {
 
   ngOnInit(): void {}
 
+  async updateTranslations(doc: Document): Promise<boolean> {
+    const translations: any = await this.readalong.getTranslations();
+    if (Object.keys(translations).length == 0) {
+      return false;
+    } else {
+      const sentence_nodes = doc.querySelectorAll("s")
+      // represents all translation nodes that have already been added
+      const translation_node_ids = Array.from(doc.querySelectorAll(".sentence__translation")).map((t_node) => t_node.id)
```

Comment:
Suggest using `Set` here instead if it isn't a problem to polyfill it (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)

Reply:
so:

```
    const sentence_nodes = doc.querySelectorAll("s")
      // represents all translation nodes that have already been added
      const translation_node_ids = new Set(Array.from(doc.querySelectorAll(".sentence__translation")).map((t_node) => t_node.id))
      sentence_nodes.forEach((sentence: HTMLElement) => {
        if (sentence.id in translations && translation_node_ids.has(sentence.id)) {
          let newSentence = document.createElement('s')
          newSentence.setAttribute('do-not-align', "true")
          newSentence.setAttribute("id", sentence.id)
          newSentence.setAttribute("class", "sentence__translation")
          newSentence.setAttribute("xml:lang", "eng")
          newSentence.innerText = translations[sentence.id]
          sentence.insertAdjacentElement('afterend', newSentence)
        }
      })
```
?

Reply:
Looks good!  A bit more readable - is the `Array.from` necessary when you create `translation_node_ids`?  Sometimes it is and sometimes it isn't, which is a pain...